### PR TITLE
release4.58.0

### PR DESCRIPTION
**Pull request by xinlian12**
Release azure-cosmos 4.58.0
azure-cosmos-spark-* 4.29.0
azure-cosmos-encryption 2.10.0

### CODE REVIEW

**Review by kushagraThapar — APPROVED** [review]
LGTM, thanks @xinlian12.

File: sdk/cosmos/azure-cosmos-spark_3_2-12/docs/quick-start.md [hunk 1]
```diff
@@ -26,16 +26,16 @@ You can use any other Spark 3.1.1 spark offering as well, also you should be abl
...
 
 For Spark 3.2:
-- Install Cosmos DB Spark Connector, in your spark Cluster [com.azure.cosmos.spark:azure-cosmos-spark_3-2_2-12:4.28.4](https://search.maven.org/artifact/com.azure.cosmos.spark/azure-cosmos-spark_3-2_2-12/4.28.4/jar)
+- Install Cosmos DB Spark Connector, in your spark Cluster [com.azure.cosmos.spark:azure-cosmos-spark_3-2_2-12:4.29.0](https://search.maven.org/artifact/com.azure.cosmos.spark/azure-cosmos-spark_3-2_2-12/4.29.0/jar)
 
 For Spark 3.3:
-- Install Cosmos DB Spark Connector, in your spark Cluster [com.azure.cosmos.spark:azure-cosmos-spark_3-3_2-12:4.28.4](https://search.maven.org/artifact/com.azure.cosmos.spark/azure-cosmos-spark_3-3_2-12/4.28.4/jar)
+- Install Cosmos DB Spark Connector, in your spark Cluster [com.azure.cosmos.spark:azure-cosmos-spark_3-3_2-12:4.29.0](https://search.maven.org/artifact/com.azure.cosmos.spark/azure-cosmos-spark_3-3_2-12/4.29.0/jar)
   
 For Spark 3.4:
```
Comment (FabianMeiswinkel) [inline sdk/cosmos/azure-cosmos-spark_3_2-12/docs/quick-start.md:37]: 3.5 is also missing - can also be added later

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM - thanks

File: sdk/cosmos/azure-cosmos-spark_3_2-12/docs/quick-start.md (hunk 1 above)
Comment (xinlian12) [reply sdk/cosmos/azure-cosmos-spark_3_2-12/docs/quick-start.md:37]: Yea Tomas has a separate PR for adding 3.5: https://github.com/Azure/azure-sdk-for-java/pull/39716

**Pull request merged**